packages/neon-core/src/u/misc.ts

```diff
@@ -78,3 +79,45 @@ export function reverseHex(hex: string): string {
   }
   return out;
 }
+
+/**
+ * Calculates the byte size of any supported input following NEO's variable int format.
+ */
+interface NeonSerializable {
+  size: number;
+  serialize: () => string;
+}
+type Serializables = number | HexString | NeonSerializable | Serializables[];
+export function getSerializedSize(value: Serializables): number {
+  switch (typeof value) {
+    case "number": {
+      if (value < 0xfd) return 1;
+      else if (value <= 0xffff) return 3;
+      else return 5;
+    }
+    case "object": {
+      if (value instanceof HexString) {
+        const size = value.byteLength;
+        return getSerializedSize(size) + size;
+      } else if (Array.isArray(value)) {
+        const array_length = value.length;
+        let size = 0;
+        if (array_length > 0) {
+          size =
+            value
+              .map((item) => getSerializedSize(item))
+              .reduce((prev, curr) => prev + curr, 0) - array_length;

```

Comment:
Why do we minus the `array_length `here?

Reply:
because for an array we should only look at the actual length of the items in the array, whereas now using the recursion we get `length prefix` + `actual length` per item.

Reply:
ah I believe the error might be then at calculating the size for a NeonSerializable. It should just be `return value.size`. `getSerializedSize(new SerializableObject())` should return 2?

Reply:
As is, it is producing exactly what it is supposed to produce
```C#
            var witness1 = new Witness
            {
                VerificationScript = new byte[] {0x01, 0x02},
                InvocationScript = new byte[] {0x03}
            };            
            
            var witness2 = new Witness
            {
                VerificationScript = new byte[] {0x01, 0x02},
                InvocationScript = new byte[] {0x03}
            };
                

            Witness[] witnesses = { witness1, witness2};

            Console.WriteLine(witnesses.GetVarSize()); // 11
```

```javascript
const w1 = new Neon.tx.Witness({verificationScript: "0102", invocationScript: "03"});
const w2 = new Neon.tx.Witness({verificationScript: "0102", invocationScript: "03"});
const witnesses = [w1, w2];
console.log(Neon.u.getSerializedSize(witnesses)); // 11
```

Reply:
The minus array_length works conveniently because for small objects, the prefix is only 1 byte big. It breaks down once you try to deserialize stuff larger than 0xfd.

```cs
            var largeByteArray = new byte[256];
            for (int i = 0; i < largeByteArray.Length; i++)
            {
                largeByteArray[i] = 0x01;
            }

            var witness1 = new Witness
            {
                VerificationScript = largeByteArray,
                InvocationScript = new byte[] { 0x03 }
            };

            var witness2 = new Witness
            {
                VerificationScript = largeByteArray,
                InvocationScript = new byte[] { 0x03 }
            };


            Witness[] witnesses = { witness1, witness2 };

            Console.WriteLine(witnesses.GetVarSize()); // 523
```

```js
    const input = [
      new Witness({
        invocationScript: "01".repeat(256),
        verificationScript: "03",
      }),
      new Witness({
        invocationScript: "01".repeat(256),
        verificationScript: "03",
      }),
    ];

    const result = getSerializedSize(input);
    expect(result).toBe(523); // 527
```

Reply:
Good catch 👍 . 
I looked at the whole use-case again and if it is going to be used at similar places as on the Python/C# side, then we don't need to be able to parse `Serializables[]` or `NeonSerializable`, only `number | HexString | NeonSerializable[]`. I've changed the code to reflect that and updated the test cases to use your example